### PR TITLE
CoCo: Bump Coco components to 0.10 releases

### PR DESCRIPTION
**Pull request by stevenhorsman**

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thans @stevenhorsman

**Review by ChengyuZhu6 — APPROVED** [review]
LGTM Thanks @stevenhorsman

File: src/agent/Cargo.toml [hunk 1]
```diff
@@ -77,7 +77,7 @@ strum = "0.26.2"
 strum_macros = "0.26.2"
 
 # Image pull/decrypt
-image-rs = { git = "https://github.com/confidential-containers/guest-components", rev = "1db6c3a87665dde58d0efa56f4e4af5fcd19620e", default-features = false, optional = true }
+image-rs = { git = "https://github.com/confidential-containers/guest-components", rev = "v0.10.0", default-features = false, optional = true }
```
Comment (Xynnn007) [inline src/agent/Cargo.toml:80]: we might also need to update to `v0.10.1` here, right?

**Comment by stevenhorsman** [comment]
Even with the updated version of alpine the rootfs-initrd is failing:
```
INFO: build directly
fetch https://mirror.math.princeton.edu/pub/alpinelinux//v3.20/main/x86_64/APKINDEX.tar.gz
6874648:error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed:ssl/statem/statem_clnt.c:1914:
ERROR: https://mirror.math.princeton.edu/pub/alpinelinux//v3.20/main: Permission denied
WARNING: Ignoring https://mirror.math.princeton.edu/pub/alpinelinux//v3.20/main: No such file or directory
ERROR: unable to select packages:
  alpine-base (no such package):
    required by: world[alpine-base]
  bash (no such package):
    required by: world[bash]
  ip6tables (no such package):
    required by: world[ip6tables]
  iptables (no such package):
    required by: world[iptables]
  libseccomp (no such package):
    required by: world[libseccomp]
Failed at 35: /sbin/apk.static -X ${MIRROR}/v${OS_VERSION}/main -U --allow-untrusted --root ${ROOTFS_DIR} --initdb add ${BASE_PACKAGES} ${EXTRA_PKGS} ${PACKAGES}
make: *** [Makefile:96: /home/runner/work/kata-containers/kata-containers/tools/packaging/kata-deploy/local-build/build/rootfs-initrd/builddir/initrd-image/.alpine_rootfs.done] Error 5
make[1]: *** [tools/packaging/kata-deploy/local-build/Makefile:72: rootfs-initrd-tarball-build] Error 2
```

**Pull request merged**